### PR TITLE
Generate extension api reference docs using library mode

### PR DESCRIPTION
**Pull request by jakolehm**
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,8 +38,8 @@
     "download:helm": "yarn run ts-node build/download_helm.ts",
     "build:tray-icons": "yarn run ts-node build/build_tray_icon.ts",
     "lint": "eslint $@ --ext js,ts,tsx --max-warnings=0  src/",
-    "mkdocs-serve-local": "yarn typedocs-extensions-api && docker build -t mkdocs-serve-local:latest mkdocs/ && docker run --rm -it -p 8000:8000 -v ${PWD}:/docs mkdocs-serve-local:latest",
-    "typedocs-extensions-api": "npx typedoc --plugin typedoc-plugin-markdown --readme none --name @k8slens/extensions --ignoreCompilerErrors --out docs/extensions/api --mode modules --excludePrivate --includes src/ src/extensions/extension-api.ts"
+    "mkdocs-serve-local": "docker build -t mkdocs-serve-local:latest mkdocs/ && docker run --rm -it -p 8000:8000 -v ${PWD}:/docs mkdocs-serve-local:latest",
+    "typedocs-extensions-api": "yarn typedoc --ignoreCompilerErrors --readme docs/extensions/typedoc-readme.md.tpl --name @k8slens/extensions --out docs/extensions/api --mode library --excludePrivate --hideBreadcrumbs --includes src/ src/extensions/extension-api.ts"
```
Comment (Nokel81) [inline package.json:42]: Do we really want to `ignoreCompilerErrors`?

File: package.json [hunk 2]
```diff
@@ -381,7 +381,7 @@
     "ts-loader": "^7.0.5",
     "ts-node": "^8.10.2",
     "type-fest": "^0.18.0",
-    "typedoc": "^0.19.2",
+    "typedoc": "0.17.0-3",
```
Comment (Nokel81) [inline package.json:384]: Why the downgrade?

File: package.json (hunk 1 above)
Comment (jakolehm) [reply package.json:42]: This PR does not change `ignoreCompilerErrors` behaviour.

File: package.json (hunk 2 above)
Comment (jakolehm) [reply package.json:384]: Library mode is only supported in this specific version. I'm still investigating if we can get similar results with newer versions.

File: package.json (hunk 1 above)
Comment (Nokel81) [reply package.json:42]: I see, still odd though.

File: package.json (hunk 2 above)
Comment (Nokel81) [reply package.json:384]: Fair (weird) but fair.

**Review by Nokel81 — APPROVED** [review]

File: yarn.lock [hunk 3]
```diff
@@ -3412,6 +3412,13 @@ babel-runtime@^6.26.0:
     core-js "^2.4.0"
     regenerator-runtime "^0.11.0"
 
+backbone@^1.4.0:
```
Comment (nevalla) [inline yarn.lock:3415]: ❤️

**Review by nevalla — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (jakolehm) [reply package.json:42]: For some reason it cannot resolve symbols like `__static` -> causing errors.

**Comment by jakolehm** [comment]
Had to downgrade typedoc markdown plugin (latest version did not produce good results with older typedoc).

@nevalla PTAL again

**Review by nevalla — APPROVED** [review]

**Pull request merged**